keps/sig-node/0034-20190130-node-os-arch-labels.md

```diff
@@ -0,0 +1,92 @@
+---
+kep-number: 34
+title: 
+authors:
+  - "@yujuhong"
+owning-sig: sig-node
+participating-sigs:
+  - sig-node
+reviewers:
+  - "@liggitt"
+approvers:
+  - "@dchen1107"
+editor: Yu-Ju Hong
+creation-date: 2019-01-30
+last-updated: 2019-01-30
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Promote Node Operating System & Archtecture labels to GA
+
+## Table of Contents
+
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+    * [Risks and Mitigations](#risks-and-mitigations)
+* [Graduation Criteria](#graduation-criteria)
+
+## Summary
+
+The kubelet has been been labeling the Node object with operating system (OS)
+and archietcture (arch) labels since Kubernetes 1.3. This proposal aims to
+promote these labels to GA and ensure a smooth transition with backward
+compatibility.
+
+Below lists the labels to promote:
+```
+beta.kubernetes.io/os
+beta.kubernetes.io/arch
+```
+
+## Motivation
+
+The labels have existed over two years and are widely in various places.
+Promoting these labels to reflect their stable status.
+
+### Goals
+
+Promote the OS and arch labels to GA.
+
+### Non-Goals
+
+Promoting any other label that is not listed in the Goals sections.
+
+## Proposal
+
+A new set of GA labels will be added:
+
+```
+kubernetes.io/os
```

Comment:
should they be `node.kubernetes.io` or does the `kubernetes.io/hostname` precedent weigh more?

Reply:
I'd tend toward graduating by just dropping `beta.`

The label is already on the node object, and the symmetry would make the transition less error prone for everyone adjusting manifests.

Reply:
> The label is already on the node object, and the symmetry would make the transition less error prone for everyone adjusting manifests.

+1. The labels have been there for a long time. Keeping the same style avoid user errors.

@thockin do you have objections for keeping the original format?